Source/JavaScriptCore/wasm/WasmEntryPlan.cpp

```diff
@@ -200,11 +198,26 @@ void EntryPlan::compileFunctions(CompilationEffort effort)
                     moveToState(State::Compiled);
                 return;
             }
+
+            if (Options::useConcurrentJIT()) {
+                // When the size of wasm binary requires 3 loops, use large limit.
+                if (m_moduleInformation->totalFunctionSize() > (3 * compileLimit * Options::numberOfWasmCompilerThreads()))
+                    compileLimit = Options::wasmLargePartialCompileLimit();
+            }
+            dataLogLnIf(WasmEntryPlanInternal::verbose, "Compile Size Limit ", compileLimit);
+
             functionIndex = m_currentIndex;
             functionIndexEnd = m_numberOfFunctions;
             for (uint32_t index = functionIndex; index < m_numberOfFunctions; ++index) {
-                bytesCompiled += m_moduleInformation->functions[index].data.size();
-                if (bytesCompiled >= Options::wasmPartialCompileLimit()) {
+                size_t byteSize = m_moduleInformation->functions[index].data.size();
+                // If One function's size is larger than the limit itself, we compile it separately from the current sequence,
+                // so that we can distribute compilation tasks more uniformly.
+                if (bytesCompiled && byteSize >= compileLimit) {
+                    functionIndexEnd = index;
+                    break;
+                }
+                bytesCompiled += byteSize;
+                if (bytesCompiled >= compileLimit) {
                     functionIndexEnd = index + 1;
```

Comment:
Why do any of this calculation if `effort == All`? We're going to loop around and ignore it anyway.

Reply:
In fact, it turned out that nobody is using `effort == All` anymore :) I'll remove this `effort` parameter. Will remove them.